bindings/CMakeLists.txt

```
@@ -50,29 +50,21 @@ endif("${SWIG_VERSION}" VERSION_LESS "${MIN_SWIG_VERSION}")
 set_source_files_properties(icub.i PROPERTIES CPLUSPLUS ON)
 
 if(CREATE_PYTHON)
+    find_package(Python REQUIRED COMPONENTS Interpreter Development)
     set(CMAKE_SWIG_FLAGS "-Wall;-module;icub;-threads")
 
-    set(ICUB_USE_PYTHON_VERSION "" CACHE STRING "Specify python version to use" )
-
-    find_package(PythonInterp ${ICUB_USE_PYTHON_VERSION} REQUIRED)
-    set(ICUB_USE_PYTHON_VERSION_ARGS)
-    if(NOT ICUB_USE_PYTHON_VERSION)
-      set (ICUB_USE_PYTHON_VERSION ${PYTHON_VERSION_STRING})
-    endif()
-    find_package(PythonLibs ${ICUB_USE_PYTHON_VERSION} EXACT)
-
-    include_directories(SYSTEM ${PYTHON_INCLUDE_DIRS})
+    include_directories(SYSTEM ${Python_INCLUDE_DIRS})
     swig_add_library(icub_python
                      LANGUAGE python
                      SOURCES icub.i)
-    target_include_directories(${SWIG_MODULE_icub_python_REAL_NAME} SYSTEM PRIVATE ${PYTHON_INCLUDE_DIRS})
-    target_link_libraries(${SWIG_MODULE_icub_python_REAL_NAME} ${PYTHON_LIBRARIES})
+    target_include_directories(${SWIG_MODULE_icub_python_REAL_NAME} SYSTEM PRIVATE ${Python_INCLUDE_DIRS})
+    target_link_libraries(${SWIG_MODULE_icub_python_REAL_NAME} ${Python_LIBRARIES})
```

If it works, the following would be more compliant to Modern CMake:

```suggestion
    target_link_libraries(${SWIG_MODULE_icub_python_REAL_NAME} Python::Python)
```

With this modification you can probably also remove the `include_directory` of line 56.